File: 品味Java的21種設計模式/dp_java/templatemethod/example1/LoginModel.java
package cn.javass.dp.templatemethod.example1;
/**
 * 描述登录人员登录时填写的信息的数据模型
 */
public class LoginModel {
	private String userId,pwd;

	public String getUserId() {
		return userId;
	}

	public void setUserId(String userId) {
		this.userId = userId;
	}

	public String getPwd() {
		return pwd;
	}

	public void setPwd(String pwd) {
		this.pwd = pwd;
	}

}

File: 品味Java的21種設計模式/dp_cpp/templatemethod/example1/LoginModel.cpp
#include "LoginModel.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace templatemethod
			{
				namespace example1
				{

					std::string LoginModel::getUserId()
					{
						return userId;
					}

					void LoginModel::setUserId(std::string userId)
					{
						this->userId = userId;
					}

					std::string LoginModel::getPwd()
					{
						return pwd;
					}

					void LoginModel::setPwd(std::string pwd)
					{
						this->pwd = pwd;
					}
				}
			}
		}
	}
}
